src/loadScriptFile.ts

```diff
@@ -11,34 +11,39 @@ import { PackageJson } from './parsers/parsePackageJson';
 import LogCategory = rpc.RpcLog.RpcLogCategory;
 import LogLevel = rpc.RpcLog.Level;
 
+let hasLoggedAttempt = 0;
+let hasLoggedWarning = false;
+
 export async function loadScriptFile(filePath: string, packageJson: PackageJson): Promise<unknown> {
     // See the following issue for more details on why we want to retry
     // https://github.com/Azure/azure-functions-nodejs-worker/issues/693
-    const fileName = path.basename(filePath);
+    const retries = 9;
     return await retry(
         async (currentAttempt: number) => {
-            if (currentAttempt > 1) {
+            if (currentAttempt > 1 && currentAttempt > hasLoggedAttempt) {
                 worker.log({
-                    message: `Retrying load of file "${fileName}". Attempt ${currentAttempt}/${10}`,
+                    message: `Retrying file load. Attempt ${currentAttempt}/${retries + 1}`,
                     level: LogLevel.Debug,
                     logCategory: LogCategory.System,
                 });
+                hasLoggedAttempt = currentAttempt;
```

Comment:
I want to log once per attempt, so that I can track how many retries it takes to solve the problem. In the old logic, I would print one log per attempt _per function_, resulting in a TON of logs for apps that have like 20 functions. So to answer your question, `hasLoggedAttempt` is greater than `currentAttempt` for the first function we retry at that attempt number, and then equal for the rest of the functions at that attempt number (hiding those logs).

Take a closer look at the Before/After logs in the description and that might help

Reply:
Got it! Yes that makes sense.